addons/mail/static/src/models/message_list_view_item.js

```diff
@@ -0,0 +1,58 @@
+/** @odoo-module **/
+
+import { registerModel } from '@mail/model/model_core';
+import { attr, one } from '@mail/model/model_field';
+import { clear } from '@mail/model/model_field_command';
+
+/**
+ * Models a relation between a message list view and a message view where
+ * message is used as iterating field.
+ */
+registerModel({
+    name: 'MessageListViewItem',
+    recordMethods: {
+        /**
+         * @private
+         * @returns {FieldCommand}
+         */
+        _computeNotificationMessageView() {
+            if (this.message.message_type === 'notification' && this.message.originThread.channel) {
+                return {};
+            }
+            return clear();
+        },
+        /**
+         * @private
+         * @returns {FieldCommand}
+         */
+        _computeMessageView() {
+            if (this.message.message_type !== 'notification' || !this.message.originThread.channel) {
+                return {};
+            }
+            return clear();
+        },
```

Comment:
Ok for just channel for the moment. As future work, would be nice to also have this nice layout for notifications in chatter. This would affect notifications like "Record created" or changes of tracking values